tools/internal/parser/parser.go

```diff
@@ -0,0 +1,536 @@
+// Package parser implements a validating parser for the PSL files.
+package parser
+
+import (
+	"net/mail"
+	"net/url"
+	"strings"
+)
+
+// Parse parses src as a PSL file and returns the parse result.
+//
+// The parser tries to keep going when it encounters errors. Parse and
+// validation errors are accumulated in the Errors field of the
+// returned File. A File with a non-empty Errors field is not a valid
```

Comment:
I just think that if you make this "A File with a non-empty Errors field is not necessarily a valid..." your point is clearer. Because it might be valid and currently you're saying it's not.

Reply:
Ah, I see. I rephrased to try and be more explicit about validity and what the caller should/should not do on a File with errors.